### PR TITLE
closes #239: Add google+ profile info to JWT token

### LINKED ISSUE #239: Google Plus User Profile
Provide the user display name and profile picture in the JWT token.

We will want to show the user's display name, email address, and profile picture in the front-end.

### PR DESCRIPTION
**Pull request by RyanBard**
* Adds call for google+ profile information to oauth service.
* Adds g+ profile information to JWT claims so it's available to the client.
* Adds g+ profile info to hmhb_user and HmhbUser for future user listing page.
* Updates g+ profile info every time a user logs in.
* Updates the unit tests.

### CODE REVIEW

File: client/src/main/webjar/authentication/authentication.config.js [hunk 1]
```diff
@@ -45,6 +45,10 @@
                 optionalUrlParams: [
                     "prompt"
                 ],
+                scope: [
```
Comment (RyanBard) [inline client/src/main/webjar/authentication/authentication.config.js]: This wasn't necessary, but being explicit on it helps reduce my need to keep jumping to satellizer's readme to look up the defaults.

File: src/main/java/org/hmhb/authentication/DefaultJwtAuthenticationService.java [hunk 2]
```diff
@@ -70,6 +70,14 @@ public String generateJwtToken(
         claims.put("admin", user.isAdmin());
         claims.put("email", user.getEmail());
         claims.put("userId", user.getId());
+        claims.put("displayName", user.getDisplayName());
```
Comment (RyanBard) [inline src/main/java/org/hmhb/authentication/DefaultJwtAuthenticationService.java]: All of this info will now be available to the client in $auth.getPayload() (after you login again -- your old token won't have this info yet).

File: src/main/java/org/hmhb/oauth/DefaultGoogleOauthService.java [hunk 3]
```diff
@@ -36,6 +42,33 @@ public GoogleTokenResponse getTokenResponse(
                     redirectUri
             ).execute();
 
+            GoogleCredential credential = new GoogleCredential()
```
Comment (RyanBard) [inline src/main/java/org/hmhb/oauth/DefaultGoogleOauthService.java]: It turns out that you're expected to make other calls for extra information :)

File: src/main/java/org/hmhb/oauth/GoogleResponseData.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * An object to hold the user's information from google's responses.
+ */
+@Immutable
+public class GoogleResponseData {
+
+    private final GoogleTokenResponse googleOauthToken;
```
Comment (RyanBard) [inline src/main/java/org/hmhb/oauth/GoogleResponseData.java]: I created this object so the service could return what it used to, and also the g+ profile info.

File: src/main/java/org/hmhb/user/DefaultUserService.java [hunk 5]
```diff
@@ -35,32 +35,32 @@ public DefaultUserService(
...
-    }
-
-    @Timed
-    @Override
-    public HmhbUser provisionNewUser(
-            @Nonnull String email
-    ) {
-        LOGGER.debug("provisionNewUser called: email={}", email);
-        requireNonNull(email, "email cannot be null");
+        user.setProfileUrl(profile.getUrl());
```
Comment (RyanBard) [inline src/main/java/org/hmhb/user/DefaultUserService.java]: This is the "always update new info from google" part.

File: src/main/java/org/hmhb/user/UserService.java [hunk 6]
```diff
@@ -2,29 +2,24 @@
...
-
-    /**
-     * Creates a new non-admin {@link HmhbUser} in the system.
-     *
-     * @param email the email to save into the {@link HmhbUser}
-     * @return the newly created {@link HmhbUser}
-     */
-    HmhbUser provisionNewUser(
-            @Nonnull String email
+    HmhbUser saveWithGoogleData(
```
Comment (RyanBard) [inline src/main/java/org/hmhb/user/UserService.java]: I'm open to suggestions on this method name.

File: src/main/resources/schema.sql [hunk 7]
```diff
@@ -11,7 +11,15 @@ DROP TABLE IF EXISTS hmhb_user;
 CREATE TABLE hmhb_user (
     id BIGSERIAL PRIMARY KEY,
     email TEXT NOT NULL UNIQUE,
-    is_admin BOOLEAN NOT NULL
+    admin BOOLEAN NOT NULL,
+    display_name TEXT,
```
Comment (RyanBard) [inline src/main/resources/schema.sql]: I'm pretty sure a few of these will always be provided (display name, image url, profile url), however, I'm not 100% sure, so I've left them all nullable.

File: src/main/java/org/hmhb/oauth/DefaultGoogleOauthService.java [hunk 8]
```diff
@@ -36,6 +42,33 @@ public GoogleTokenResponse getTokenResponse(
...
+
+            Plus plus = new Plus.Builder(
+                    new NetHttpTransport(),
+                    new JacksonFactory(),
+                    new HttpRequestInitializer() {
+                        @Override
+                        public void initialize(HttpRequest request) throws IOException {
+                            // empty
+                        }
+                    }
```
Comment (mckoon) [inline src/main/java/org/hmhb/oauth/DefaultGoogleOauthService.java]: Looks like Google's Builder needs a Builder...

Comment (RyanBard) [inline src/main/java/org/hmhb/oauth/DefaultGoogleOauthService.java]: Some of them have the, others don't (or maybe it depends on library versions or something).

File: src/main/java/org/hmhb/user/UserService.java (hunk 6 above)
Comment (mckoon) [inline src/main/java/org/hmhb/user/UserService.java]: It's fine

**Pull request merged**